apps/site/src/_pages/docs/4_spec/02_graph-module.mdx

```diff
@@ -1,303 +1,445 @@
+    - MAY contain a field `rightToLeftOrder` used to order this link entity against other link entities of the same type with the same `rightEntityId`, which MUST be a non-negative integer
 
-**Links** represent connections or relations between entities. In the language of an entity graph, the links are the edges, and the entities are the nodes.
+i.e. any entity MUST conform to the following schema:
 
-**Linked aggregations** represent a link from a block to a particular aggregation of entities, e.g. the top 10 entities of type Person, sorted by a particular property.
+<EntityMetaSchema />
 
-#### <Hidden>Structure </Hidden>Specification
+### <Hidden>Entity </Hidden>Identification
 
-##### Entities
+#### <Hidden>Entity Identification </Hidden>Summary
 
-When embedding applications pass entities to blocks, whether the **block entity** or any other:
+Blocks must pass an `entityId` when making requests in respect of any entity.
 
-- entities MUST be represented by an object, which contains one or more **identifying fields** at the top level. The object:
-  - MUST contain an `entityId` identifying the entity, which MUST be a string.
-  - MAY contain an `entityTypeId` identifying the entity type the entity belongs to, which MUST be a string (if defined).
-- MUST contain a `properties` field if the block has defined any `required` properties in its schema, and otherwise MAY be omitted. If present, it:
-  - MUST be an object
-  - MUST conform to the constraints described in the entity’s schema, if any
+Embedding applications should ensure this is stable and unique for any given entity.
 
-i.e. any entity MUST conform to the following schema:
+#### <Hidden>Entity Identification </Hidden>Specification
+
+Blocks MUST provide `entityId` when identifying a particular entity for the purpose of **messages** under the graph module.
+
+Embedding applications MUST provide `entityId` for each entity it supplies the block at the root of the entity data object, as described under [Structure](#structure) above.
+
+`entityId` SHOULD be a stable, unique reference to the entity, so that blocks may use it for comparing the identity of entities – i.e. so that blocks can assume that entities with the same `entityId` are the same entity, and that the same entity referenced in different places will have the same `entityId` wherever it appears.
+
+## Subgraph Definition
+
+### <Hidden>Subgraph </Hidden>Summary
+
+Many of the values returned by embedding applications for messages of the graph module include a **subgraph**.
+
+A subgraph is a data structure which contains elements of the graph, **vertices**, and **edges** which is a precomputed lookup set of the relationships between elements of the graph.
+Consumers can pass this object to the graph module package's standard library to benefit from implementations of common operations such as getting entities linked to a particular entity.
```

Comment:
I _think_ this is a bit out of place as an implementation detail. Could be convinced otherwise
```suggestion
```

Reply:
Will remove, I had written this because I was going to try and avoid defining the `Subgraph` and instead say that the type's shape wasn't explicitly defined but it should be consumable by the standard library or something.

Ended up defining it anyway